src/network/address.rs

```diff
@@ -118,5 +146,16 @@ mod test {
                              0, 0, 0, 0, 0, 0, 0, 0, 0, 0xff, 0xff, 0x0a, 0, 0, 1]);
         assert!(addr.is_err());
     }
+
+    #[test]
+    fn test_socket_addr () {
+        let s4 = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(111,222,123,4)), 5555);
+        let a4 = Address::new(&s4, 9);
```

Comment:
Can you add a test that this mapps correctly to bitcoin's ipv6-mapped addresses? I assume its correct, but good to check as IIRC there are multiple ways to map (I belive Bitcoin uses the ::ffff:IPv4 one, but should double-check).

Reply:
That is what Ipv6Addr::to_ipv4 checks and I used in Address::socket_addr() therefore the test should never fail.

Reply:
see in deserialize_address_test